.github/workflows/CI-pr.yml

```diff
@@ -351,10 +455,155 @@ jobs:
 
           for package in "${PACKAGES_CHECKED[@]}"; do
             printf '::group::Running `cargo hack` on package `%s`\n' "${package}"
-            cargo hack check --each-feature --all-targets --package "${package}"
+            cargo hack check --each-feature --all-targets --package "${package}" --exclude-features 'v2 merchant_account_v2 payment_v2'
             echo '::endgroup::'
           done
 
+
+  test-v2:
+    name: Run tests on stable toolchain for v2 related features
+    runs-on: ${{ matrix.runner }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        runner:
+          - ubuntu-latest
+          - hyperswitch-runners
+
+    steps:
+      - name: Generate a token
+        if: ${{ github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name }}
+        id: generate_token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ secrets.HYPERSWITCH_BOT_APP_ID }}
+          private-key: ${{ secrets.HYPERSWITCH_BOT_APP_PRIVATE_KEY }}
+
+      - name: Checkout repository for fork
+        if: ${{ (github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name) }}
+        uses: actions/checkout@v4
+
+      - name: Checkout repository with token
+        if: ${{ (github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name) }}
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+          token: ${{ steps.generate_token.outputs.token }}
+
+      - name: "Fetch base branch"
+        shell: bash
+        run: git fetch origin $GITHUB_BASE_REF --depth 1
+
+      - name: Install mold linker
+        uses: rui314/setup-mold@v1
+        if: ${{ runner.os == 'Linux' }}
+        with:
+          make-default: true
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable 2 weeks ago
+          components: clippy
+
+      - name: Install sccache
+        uses: taiki-e/install-action@v2.33.28
+        with:
+          tool: sccache
+          checksum: true
+
+      - name: Install cargo-hack
+        uses: taiki-e/install-action@v2.33.28
+        with:
+          tool: cargo-hack
+          checksum: true
+
+      # - name: Install cargo-nextest
+      #   uses: taiki-e/install-action@v2.33.28
+      #   with:
+      #     tool: cargo-nextest
+      #     checksum: true
+
+      # - name: Setup Embark Studios lint rules
+      #   shell: bash
+      #   run: |
+      #     mkdir -p .cargo
+      #     curl -sL https://raw.githubusercontent.com/EmbarkStudios/rust-ecosystem/main/lints.toml >> .cargo/config.toml
```

Comment:
What is this, are we using this anywhere else?

Reply:
Some of the lints that we have in our [`.cargo/config.toml`](https://github.com/juspay/hyperswitch/blob/34d2cec151d2ceeb51a638a661bc1a4c716b2859/.cargo/config.toml) file have been picked from there.